newIDE/app/src/MainFrame/index.js

```diff
@@ -2273,6 +2279,26 @@ const MainFrame = (props: Props) => {
     editorTab: EditorTab,
     newState: State = state
   ) => {
+    if (
```

Comment:
That's indeed not super pretty/fragile. I don't like these new refs... 
Can we check why this code is not working as it should?

<img width="555" alt="image" src="https://user-images.githubusercontent.com/1280130/225088625-ffb2459b-29bf-4973-9ab9-e0e1b8867f46.png">

in newIDE/app/src/MainFrame/EditorContainers/EventsFunctionsExtensionEditorContainer.js.

Reply:
I wanted to avoid the `componentWillUnmount` to call `onLoadEventsFunctionsExtensions`, I tried to:
- call `onLoadEventsFunctionsExtensions` when another tab is put upfront after an extension tab has been left
- Disable both `componentWillUnmount` (to avoid it on project closing) and `componentDidUpdate` (because it's already done)

The big issue is that the Home tab is handled differently because it's what will be shown when a project is closed. Still, users could open Home and launch a preview and I'm not sure the last edited extension would be up to date.

I'm not sure a clean solution can be done with only UI events. I think we should invest in a dirty flag system which will allow finer optimizations.